### ISSUE TITLE
Type with symbol parameter doesn't save correctly in JLD

### ISSUE THREAD

**Issue by GunnarFarneback** [timholy/HDF5.jl#211]
This fails:
```julia
julia> using HDF5, JLD

julia> type Foo{T} end

julia> a = Foo{:a}()
Foo{:a}()

julia> save("test.jld", "a", a)

julia> b = load("test.jld", "a")
WARNING: type Foo{a} not present in workspace; reconstructing
##Foo{a}#7932()

julia> @assert typeof(a) == typeof(b)
ERROR: assertion failed: typeof(a) == typeof(b)
 in error at ./error.jl:21
```
The core problem seems to be that symbols print without any indication of being symbols.
```julia
julia> print(:a)
a
```
One possible solution is to add a method
```julia
full_typename(io::IO, ::JldFile, x::Symbol) = print(io, ":", x)
```
in JLD.jl but I don't know whether that's the right way to go.

**Comment by mbauman** [comment]
Sorry I didn't notice this sooner - This was fixed in #203 (with exactly your solution), but a new release of HDF5 hasn't been tagged yet.

**Comment by mbauman** [comment]
It may not end up being the final solution here, however.  See #204.
